pkg/apis/serving/v1alpha1/kfservice_framework.go

```diff
@@ -32,15 +32,19 @@ const (
 	ExactlyOneModelSpecViolatedError = "Exactly one of [Custom, Tensorflow, SKLearn, XGBoost] must be specified in ModelSpec"
 	// AtLeastOneModelSpecViolatedError is a known error message
 	AtLeastOneModelSpecViolatedError = "At least one of [Custom, Tensorflow, SKLearn, XGBoost] must be specified in ModelSpec"
+
+	TensorflowServingImageConfigName = "TensorflowServingImage"
+	XgboostServingImageConfigName    = "XgboostServingImage"
+	SklearnServingImageConfigName    = "SklearnServingImage"
 )
 
 var (
 	DefaultMemoryRequests = resource.MustParse("2Gi")
 	DefaultCPURequests    = resource.MustParse("1")
 )
 
-func (m *ModelSpec) CreateModelServingContainer(modelName string) *v1.Container {
-	return getHandler(m).CreateModelServingContainer(modelName)
+func (m *ModelSpec) CreateModelServingContainer(modelName string, configs map[string]string) *v1.Container {
```

Comment:
High level comment. I was looking at this. Do you think the best way is to pass around this config? Do you think it might be better to have a global configmap object that's readable? It would be a little challenging with the reconciliation pattern.

How does istio/knative/kubeflow do this?

Reply:
ye I thought global ConfigMap would be hard to reconcile, I will check how knative does this.

Reply:
Here's a thought. Perhaps a little beyond this scope. What if we had a set of "reconcilers" that we instantiated.

We could pass in:
1. A k8s api client
2. a config map

Whenever the configmap changed, we'd need to reinstantiate all of our controllers, but it might result in a cleaner flow.